Put AccountSequence in SignerInfo (#6997)
cosmos/cosmos-sdk: docs/architecture/adr-020-protobuf-transaction-encoding.md

addressed in a future ADR, but it should build off of these proposals.

Based on detailed discussions ([\#6030](https://github.com/cosmos/cosmos-sdk/issues/6030)

> and [\#6078](https://github.com/cosmos/cosmos-sdk/issues/6078)), the original

Author: sorry, spacing changes are made by my editor... but i think they make sense.

Reviewer: Thank you editor. By the way, have you considered adding an .editorconfig to this repo? I made very good experience with it and use it in all projects: https://editorconfig.org/

    // ModeInfo describes the signing mode of the signer and is a nested
    // structure to support nested multisig pubkey's
    ModeInfo mode_info = 2;
    // account_sequence is the sequence of the account, which describes the

>     // number of transactions sent from a given address. It is used to prevent

Reviewer: ```suggestion
    // number of transactions signed by a given address. It is used to prevent
```

Author: While "sent from" might not be perfect, I still believe it's clearer than "signed by". I can sign 100 transactions, and never broadcast them.

Reviewer 2: And I can sent multisig transactions I never signed (I have done this a few time IRL). How about:

`// number of committted transactions signed by a given address`

"committed" should be understood to be in a block (tendermint commits the block), but maybe there is a better word

>     // replay attacks.

Reviewer: Is there really no way to change the sequence other than signing a transaction?

Author: What other ways do you have in mind?

For me, sequence == number of transactions in all past blocks signed by an address.

Reviewer 2: That is true, it is incremented by 1 for every transaction that is accepted into a block with a valid signature from this account.

Some of those transactions may be errors and not changing state, but still deduct fees and update the sequence number. But all of those are in a block with a valid signature (it is gets rejected due to low fees, it doesn't make it in the block)

Reviewer: I agree this is the default case, but is this guaranteed now and in the future? There can be a way to bump the sequence in order to invalidate a range of sequence numbers.

E.g. I sign a 20 transactions to be broadcasted in the future. Then I decide I don't want them to be executed. In order to do that, I send a bump sequence message to the chain which bumps my sequence by let's say 30. Then all those signatures are invalid.

Reviewer 3: This is the way it is now and has always been. I don't see any means of change this behavior any time soon.

>     uint64 account_sequence = 3;

Reviewer: As above, I suggest  `account_sequence` -> `sequence`
